code/game/machinery/deployable.dm

```diff
@@ -254,24 +81,79 @@ for reference:
 		var/obj/item/projectile/proj = mover
 		if(proj.firer && Adjacent(proj.firer))
 			return 1
-		if(prob(20))
+		if(prob(proj_pass_rate))
 			return 1
 		return 0
 	else
 		return 0
 
-/obj/machinery/deployable/barrier/proc/explode()
 
-	visible_message("<span class='danger'>[src] blows apart!</span>")
-	var/turf/Tsec = get_turf(src)
 
-/*	var/obj/item/stack/rods/ =*/
-	new /obj/item/stack/rods(Tsec)
 
-	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
-	s.set_up(3, 1, src)
-	s.start()
 
-	explosion(src.loc,-1,-1,0)
-	if(src)
-		qdel(src)
+/////BARRICADE TYPES///////
+
+/obj/structure/barricade/wooden
+	name = "wooden barricade"
+	desc = "This space is blocked off by a wooden barricade."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "woodenbarricade"
+
+
+/obj/structure/barricade/security
+	name = "security barrier"
+	desc = "A deployable barrier. Provides good cover in fire fights."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrier0"
+	density = 0
+	anchored = 0
+	health = 180
+	maxhealth = 180
+	proj_pass_rate = 20
+	ranged_damage_modifier = 0.5
+
+
+/obj/structure/barricade/security/New()
+	..()
+	spawn(40)
+		icon_state = "barrier1"
+		density = 1
+		anchored = 1
+		visible_message("<span class='warning'>[src] deploys!</span>")
+
+
+/obj/item/weapon/grenade/barrier
+	name = "barrier grenade"
+	desc = "Instant cover. Alt+click to toggle modes."
+	icon = 'icons/obj/grenade.dmi'
+	icon_state = "flashbang"
+	item_state = "flashbang"
+	var/mode = SINGLE
+
+/obj/item/weapon/grenade/barrier/AltClick(mob/user)
+	..()
+	switch(mode)
+		if(SINGLE)
+			mode = VERTICAL
+		if(VERTICAL)
+			mode = HORIZONTAL
+		if(HORIZONTAL)
+			mode = SINGLE
+
+	user << "[src] is now in [mode] mode."

```

Comment:
Could add an action button for this as well, just for completeness and intuitiveness.